dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/DatasetSaver.java

```diff
@@ -35,26 +34,37 @@
 
     private static final Logger logger = Logger.getLogger(DatasetPage.class.getCanonicalName());
     
-    @EJB
+    
     private EjbDataverseEngine commandEngine;
-    @EJB
     private UserNotificationService userNotificationService;
-    @EJB
     private DatasetServiceBean datasetService;
-    @Inject
     private DataverseSession session;
-    @Inject
     private DataverseRequestServiceBean dvRequestService;
-    @EJB
     private IngestServiceBean ingestService;
-    @EJB
     private SettingsServiceBean settingsService;
-    @Inject
     private ProvPopupFragmentBean provPopupFragmentBean;
+    
+    
+    // -------------------- CONSTRUCTORS --------------------
```

Comment:
I think it will blowup without default constructor for EJB proxy.

Reply:
Hmm, not sure why but I run glassfish and it wasn't any problem with it.
I would have to dig more deeply when no-arg constructor is needed and when not.
My guess is it have something to do with implementing `Serializable` (this class do not implement it).
You think that we shoud add no-arg constructor just in case?

Reply:
Hmm, if it's working fine here I would leave it be. But I will also have to dive deeper when it is needed